Account/FxAClient10.swift

```diff
@@ -659,32 +651,35 @@ open class FxAClient10 {
 
     fileprivate func makeRequest<T>(_ request: URLRequest, responseHandler: @escaping (JSON) -> T?) -> Deferred<Maybe<T>> {
         let deferred = Deferred<Maybe<T>>()
+        urlSession.dataTask(with: request) { (data, response, error) in
```

Comment:
It seems like there's a lot of code in this block here that would be common anytime we fetch JSON from an HTTP API.

It might be useful to just make a helper function in your utils file below that contains:
```
func fetchJSON(withSession: URLSession, request: URLRequest) -> Deferred<Maybe<JSON>>
```

...where `fetchJSON()` automatically invokes the `resume()` call on the task and handles validation of the response. Also, you could use a `CancellableDeferred` where cancellation invokes `cancel()` on the task.

Reply:
I was hoping to keep this PR as minimal as possible, could that be a follow up for more refactoring? I feel like the more refactoring, the greater the risk of breakage.